### DISCUSSION TITLE
Support _app directory to act as _app.js

### DISCUSSION THREAD

**Discussion by msafi** [vercel/next.js#35361] [Ideas]
### Describe the feature you'd like to request

Organizing files of `pages` inside directories helps. The problem is that `_app` can't be a directory. So you end up with this situation

```
.
└── pages/
    ├── somePage/
    │   ├── index.js
    │   ├── somePageComponent1.js
    │   ├── somePageComponent2.js
    │   └── etc.js
    ├── anotherPage/
    │   ├── index.js
    │   └── ...etc
    ├── _app.js
    └── _appComponent(where do I live? 😔).js
```

### Describe the solution you'd like

Ideally, it should be possible to make `_app` a directory so that supporting components can live within it as well, like this

```
.
└── pages/
    ├── somePage/
    │   ├── index.js
    │   ├── somePageComponent1.js
    │   ├── somePageComponent2.js
    │   └── etc.js
    ├── anotherPage/
    │   ├── index.js
    │   └── ...etc
    └── _app/
        ├── index.js
        └── _appComponent(This is my home 😊).js
```

### Describe alternatives you've considered

Have an `_app` directory that `_app.js` imports from.

**Comment by kristijorgji** [comment]
_appComponent.j|tsx can live anywhere.
In my project I put them outside pages folder, you can have components folder outside pages and store all things there.

**Comment by msafi** [comment]
Thanks for the suggestion @kristijorgji. The organization I was most satisfied with was putting all my source code in a `src/` directory and doing nothing more than export on the files in `pages/`.

So the files are structured like this

```
pages
  _app.ts
  foo.ts
  index.ts
src
  _app
    index.tsx
    _appComponent1.tsx
    _appComponent2.tsx
  foo
    index.tsx
    fooComponent1.tsx
    ...etc
  ...etc
```

Then on `pages/foo.ts` I would only do an export from `src/foo`, like this

```ts
// pages/foo.ts
export { Foo as default } from '../src/foo';
export { getStaticPaths } from '../src/foo/getStaticPaths';
export { getStaticProps } from '../src/foo/getStaticProps';
```
